### PR TITLE
[Serverless] Fix trace loss

### PR DESCRIPTION
**Pull request by maxday**
### What does this PR do?

When a huge amount of invocation occurs in a very short period of time, traces could be lost.

This PR includes : 
- change the final waiting time for logs from 1s to 50ms (as log are sent each 25ms, no need to wait 1s, since we only have 2 sec to perform the final flush)
- refactoring out complex waitGroup logic to introduce an `orchestrator`
- block receiving traces outside of the invocation event lifecycle

### Motivation

fix trace loss

loadTestResult
loadTest

### Describe how to test/QA your changes

unit tests

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/api.go [hunk 1]
```diff
@@ -84,6 +84,10 @@ type HTTPReceiver struct {
 
 	// outOfCPUCounter is counter to throttle the out of cpu warning log
 	outOfCPUCounter *atomic.Uint32
+
+	// PreHookFunc is a function executed when a payload is received from a tracer
+	// used in serverless context to block until ready to properly handle the payload
+	PreHookFunc func()
```
Comment (knusbaum) [inline pkg/trace/api/api.go:90]: Let's call this maybe `onHandleTraces`.

**Review by knusbaum — CHANGES_REQUESTED** [review]
See inline comments

File: pkg/trace/api/api.go (hunk 1 above)
Comment (maxday) [reply pkg/trace/api/api.go]: I capitalized it since it needs to be accessible from a serverless package

**Review by knusbaum — APPROVED** [review]
This can me merged once conflicts are resolved.

**Pull request closed without merging**